src/main/scala/org/ergoplatform/nodeView/ErgoInterpreter.scala

```diff
@@ -0,0 +1,61 @@
+package org.ergoplatform.nodeView
+
+import org.ergoplatform.ErgoLikeContext.Height
+import org.ergoplatform._
+import org.ergoplatform.settings.{Constants, Parameters}
+import sigmastate.SBoolean
+import sigmastate.Values.Value
+import sigmastate.interpreter.Interpreter.VerificationResult
+import scala.util.Try
+
+
+class ErgoInterpreter(override val maxCost: Long = Parameters.MaxBlockCost)
+  extends ErgoLikeInterpreter(maxCost) {
+
+  override type CTX = ErgoContext
+
+  /**
+    * Check that expired box is spent in a proper way
+    *
+    * @param box    - box being spent
+    * @param output - newly created box
+    * @param currentHeight - current height of the blockchain (at the moment of spending)
+    * @return whether the box is spent properly according to the storage fee rule
+    */
+  protected def checkExpiredBox(box: ErgoBox, output: ErgoBoxCandidate, currentHeight: Height): Boolean = {
+    val maxStorageFee = Parameters.K * box.bytes.length * (output.creationHeight - box.creationHeight)
+
+    (box.value - maxStorageFee <= 0 && box.value == output.value) || {
+      output.creationHeight == currentHeight &&
+        output.value >= box.value - maxStorageFee &&
+        ErgoBox.allRegisters.tail.forall(rId => rId == ErgoBox.ReferenceRegId || box.get(rId) == output.get(rId))
+    }
+  }
+
+  override def verify(exp: Value[SBoolean.type],
+                      context: CTX,
+                      proof: Array[Byte],
+                      message: Array[Byte]): Try[VerificationResult] = {
+
+    lazy val varId = Constants.StorageIndexVarId
+
+    //no proof provided and enough time since box creation to spend it
+    if (context.currentHeight - context.self.creationHeight >= Constants.StoragePeriod
+      && proof.length == 0
+      && context.extension.values.contains(varId)) {
+
+      Try {
+        val idx = context.extension.values(varId).value.asInstanceOf[Short]
+        val outputCandidate = context.spendingTransaction.outputCandidates(idx)
```

Comment:
What about the case, when box value is not enough to pay storage fee and box should be just removed without  new output creation?

Reply:
I think it should be `Option[BoxCandidate]` here, that is None if box is just destroyed

Reply:
Line 28: "(box.value - maxStorageFee <= 0)". I think Boolean return value type is enough for checkExpiredBox. In future, "context.extension.values.contains(varId)" on line 45 is not needed when storage fee exceeds box value, but for now (while lazy evaluation is not done) we can not avoid it.